idpf_specification.md

```diff
@@ -6433,28 +6433,101 @@ VIRTCHNL2_OP_RDMA			= 529,
+##### Cross time stamping flow done on all CPU complexes attached to the device.
+1.	If the OS supports hooks for PCIE based PTM capability and a callback is received to do PTM cross time stamping, if the capability was granted to the driver instance, the driver makes a call to either do a direct or mailbox based PTM cross time stamping request to the device. Direct method is preferred over the mailbox method for cross timestamping to avoid unnecessary mailbox chatter. The opcodes used for indirect PTM access is VIRTCHNL2_OP_PTP_GET_CROSS_TIME
+2.	If the capability was not granted the driver will fail the call.
+3.	Legacy method of cross timestamping is pretty much supported in all modern OSes although may not be invoked in case of a VM. VM system time may be synchronized with the Hypervisor or Host OS time through a SW flow. Direct access is the only right method for this, mailbox mechanism is only for completeness. 
+##### Tx Timestamping Flow (given to drivers that are not untrusted(tenant) drivers.)
+a.	In this case if the device control plane grants the direct or indirect access for Tx timestamp, the driver learns about the number of latches and the index of the latches as part of initialization. If the access grated is direct access, the driver also learns about the register offsets of the latches in the PCIE MMIO space.
+b.	When a packet comes down with a PTP time stamping request as part of packet meta data from the stack then the driver must enable the Tx timestamping bits in the Tx descriptor and specify one of the free/unused latch ID to be used for timestamping of this Tx packet by the HW.
+c.	 At some point when the packet completion for both buffer and the Tx descriptor is received, then the driver can go request the Tx timestamp reading directly or indirectly over the mailbox. Note:  Here the packet buffer completion is assumed to happen very close to packet scheduling on the wire by the device with no buffering of the packet after that point in the device. The opcode used for indirect tx timestamping reading is VIRTCHNL2_OP_PTP_GET_VPORT_TX_TSTAMP	
+d.	When reading the timestamp over mailbox, occasionally the driver can combine the request to get the device time as well along with the Tx timestamp instead of requesting just the timestamp. This is just an optimization to avoid having two mailbox request and cause jitter.
+e.	The driver can now return the latch ID to the free pool, to be used for another PTP timestamping packet.
+##### Rx Timestamping flow
+	For now every 32 byte default RX descriptor carrying a valid packet whether in Single queue or split queue mode is assumed to carry RX timestamp for when the packet is received in the device (mostly at PHY level). A valid rx timestamp value is indicated by device by setting the lowest bit to 1.
```

Comment:
- nitpick: unnecessary indentation
- nitpick: upper case 'Single'
- nitpick: 'Rx' instead of 'RX' and 'rx'

Reply:
Every packet fo the flow carries an Rx timestamp. So there is no flow granularity.